File: app/src/main/java/org/scp/gymlog/room/DBThread.java
package org.scp.gymlog.room;

import android.content.Context;

import java.util.function.Consumer;

public class DBThread {
    public static void run(Context context, Consumer<AppDatabase> process) {
        AppDatabase database = Connection.get(context);
        new Thread(()->
            process.accept(database)
        ).start();
    }
}

File: app/src/main/java/org/scp/gymlog/room/DBThread.kt
package org.scp.gymlog.room

import android.content.Context
import java.util.function.Consumer

object DBThread {
    fun run(context: Context, process: Consumer<AppDatabase>): Thread {
        val database = Connection[context]
        return Thread { process.accept(database) }.apply {
            start()
        }
    }
}